Branch-SDK/src/main/java/io/branch/referral/util/BranchEvent.java

```diff
@@ -244,16 +244,48 @@ public String getEventName() {
      * Logs this BranchEvent to Branch for tracking and analytics
      *
      * @param context Current context
-     * @return {@code true} if the event is logged to Branch
+     * @return {@code true}
      */
     public boolean logEvent(Context context) {
-        boolean isReqQueued = false;
+        logEvent(context, null);
+        return true;
+    }
+
+    public interface BranchLogEventCallback {
+        void onEventLogged(boolean success);
+        void onFailure(int errorCode, String errorMessage);
+    }
+
+    /**
+     * Logs this BranchEvent to Branch for tracking and analytics and provides a callback once the event is logged
+     *
+     * @param context  Current context
+     * @param callback A {@link BranchLogEventCallback} callback that gets triggered once the event is logged
+     */
+    public void logEvent(Context context, final BranchLogEventCallback callback) {
         Defines.RequestPath reqPath = isStandardEvent ? Defines.RequestPath.TrackStandardEvent : Defines.RequestPath.TrackCustomEvent;
         if (Branch.getInstance() != null) {
-            Branch.getInstance().handleNewRequest(new ServerRequestLogEvent(context, reqPath, eventName, topLevelProperties, standardProperties, customProperties, buoList));
-            isReqQueued = true;
+            Branch.getInstance().handleNewRequest(
+                    new ServerRequestLogEvent(context, reqPath, eventName, topLevelProperties, standardProperties, customProperties, buoList) {
+                        @Override
+                        public void onRequestSucceeded(ServerResponse response, Branch branch) {
+                            if (callback != null) {
+                                callback.onEventLogged(true);
```

Comment:
Instead of passing `true`, pass the response and Branch object back to the caller.